var/spack/repos/builtin/packages/cgal/package.py

```diff
@@ -59,28 +65,52 @@ class Cgal(CMakePackage):
     depends_on('qt@5:', when='+demos')
 
     # Optional Third Party Libraries
+    depends_on('eigen', when='+eigen')
+
     # depends_on('leda')
     # depends_on('mpfi')
     # depends_on('rs')
     # depends_on('rs3')
     # depends_on('ntl')
-    # depends_on('eigen')
     # depends_on('libqglviewer')
     # depends_on('esbtl')
     # depends_on('intel-tbb')
 
+    # Header only builds became optimal in 4.9, default thereafter
+    conflicts('~header_only', when='@:4.9')
+
+    def setup_build_environment(self, env):
+        spec = self.spec
+
+        env.set('BOOST_INCLUDEDIR', spec['boost'].headers.directories[0])
+        env.set('BOOST_LIBRARYDIR', spec['boost'].prefix.lib64)
```

Comment:
```suggestion
        env.set('BOOST_LIBRARYDIR', spec['boost'].libs.directories[0])
```

Reply:
Is there a doc page somewhere describing which of the two options is preferred ? I've checked [this](https://spack.readthedocs.io/en/latest/packaging_guide.html#accessing-dependencies) but it doesn't list all the members of the spec object.